### ISSUE TITLE
[Suggestion] Adding MangaUpdates.com as a source

### ISSUE THREAD

**Issue by ShrinkWrapper** [diogovalentte/mantium#1]
Been using mantium for a while now and find it very useful for tracking my mangas!

Was wondering if it would be possible to add mangaupdates.com as a source at some point?
The way it differs from the current sources is that it doesn't actually host any manga. Its more of a database/wiki like imdb making it a bit more proper then relaying on the metadata off piracy sites.
They offer an API as well for browsing the database. https://api.mangaupdates.com

This could perhaps be implemented as the base form of adding sources in mantium without having to rely on trackers and adding urls.

Example of API result for the manga Heavenly Delusion:

GET: https://api.mangaupdates.com/v1/series/45702686526

```
{
  "series_id": 45702686526,
  "title": "Tengoku Daimakyou",
  "url": "https://www.mangaupdates.com/series/kzu76m6/tengoku-daimakyou",
  "associated": [
    {
      "title": "Heavenly Delusion"
    },
    {
      "title": "O Paraíso Ilusório"
    },
    {
      "title": "Złudne niebo (Polish)"
    },
    {
      "title": "Великая небесная стена"
    },
    {
      "title": "Иллюзия рая"
    },
    {
      "title": "توهم بهشتی"
    },
    {
      "title": "وَهْم السماء"
    },
    {
      "title": "天国大魔境"
    },
    {
      "title": "천국대마경"
    }
  ],
  "description": "From Denpa:<BR>Within the safety of the walls, youths are raised in a nursery-style setting by robots. While life there may appear stale on the surface, the children are full of potential and curiosity. In many ways, it is like a slice of heaven. The outside world is a hellscape. It is almost entirely void of anything mechanical and is now inhabited by bizarre, yet powerful supernatural beings.<BR>Maru, with the aid of Kiruko, is out there crisscrossing what was once Tokyo for heaven. But after searching for so long, maybe heaven is more of an untenable dream than a potential reality.<BR><BR>Note: Winner of Kono Manga ga Sugoi! 2019 Award in Male Readers category.",
  "image": {
    "url": {
      "original": "https://cdn.mangaupdates.com/image/i443117.png",
      "thumb": "https://cdn.mangaupdates.com/image/thumb/i443117.png"
    },
    "height": 228,
    "width": 160
  },
  "type": "Manga",
  "year": "2018",
  "bayesian_rating": 7.45,
  "rating_votes": 141,
  "genres": [
    {
      "genre": "Action"
    },
    {
      "genre": "Adventure"
    },
    {
      "genre": "Drama"
    },
    {
      "genre": "Gender Bender"
    },
    {
      "genre": "Horror"
    },
    {
      "genre": "Mature"
    },
    {
      "genre": "Mystery"
    },
    {
      "genre": "Psychological"
    },
    {
      "genre": "Sci-fi"
    },
    {
      "genre": "Seinen"
    },
    {
      "genre": "Tragedy"
    }
  ],
  "categories": [
    {
      "series_id": 45702686526,
      "category": "Adapted to Anime",
      "votes": 4,
      "votes_plus": 4,
      "votes_minus": 0,
      "added_by": 16798809483
    },
    {
      "series_id": 45702686526,
      "category": "Appearance Different from Personality",
      "votes": 3,
      "votes_plus": 3,
      "votes_minus": 0,
      "added_by": 54489895811
    },
    {
      "series_id": 45702686526,
      "category": "Award-Winning Work",
      "votes": 9,
      "votes_plus": 9,
      "votes_minus": 0,
      "added_by": 65222243716
    },
    {
      "series_id": 45702686526,
      "category": "Blood and Gore",
      "votes": 8,
      "votes_plus": 8,
      "votes_minus": 0,
      "added_by": 17832369998
    },
    {
      "series_id": 45702686526,
      "category": "Body Modification",
      "votes": 9,
      "votes_plus": 9,
      "votes_minus": 0,
      "added_by": 17832369998
    },
    {
      "series_id": 45702686526,
      "category": "Bodyguard/s",
      "votes": 10,
      "votes_plus": 9,
      "votes_minus": 1,
      "added_by": 77054585918
    },
    {
      "series_id": 45702686526,
      "category": "Child/ren",
      "votes": 10,
      "votes_plus": 10,
      "votes_minus": 0,
      "added_by": 70020080814
    },
    {
      "series_id": 45702686526,
      "category": "Controversial",
      "votes": 3,
      "votes_plus": 2,
      "votes_minus": 1,
      "added_by": 53789420519
    },
    {
      "series_id": 45702686526,
      "category": "Dark Ambience",
      "votes": 1,
      "votes_plus": 1,
      "votes_minus": 0,
      "added_by": 2095673730
    },
    {
      "series_id": 45702686526,
      "category": "Hermaphrodite",
      "votes": 0,
      "votes_plus": 0,
      "votes_minus": 0,
      "added_by": 23459618639
    },
    {
      "series_id": 45702686526,
      "category": "Human Experiment/s",
      "votes": 8,
      "votes_plus": 8,
      "votes_minus": 0,
      "added_by": 44221078641
    },
    {
      "series_id": 45702686526,
      "category": "Identity Crisis",
      "votes": 2,
      "votes_plus": 2,
      "votes_minus": 0,
      "added_by": 58695495412
    },
    {
      "series_id": 45702686526,
      "category": "Japan",
      "votes": 2,
      "votes_plus": 2,
      "votes_minus": 0,
      "added_by": 2095673730
    },
    {
      "series_id": 45702686526,
      "category": "Journey",
      "votes": 4,
      "votes_plus": 4,
      "votes_minus": 0,
      "added_by": 54489895811
    },
    {
      "series_id": 45702686526,
      "category": "Kidnapping/s",
      "votes": 1,
      "votes_plus": 1,
      "votes_minus": 0,
      "added_by": 2095673730
    },
    {
      "series_id": 45702686526,
      "category": "LGBT Character in Non-Yaoi/Yuri Manga",
      "votes": 9,
      "votes_plus": 9,
      "votes_minus": 2,
      "added_by": 12594761460
    },
    {
      "series_id": 45702686526,
      "category": "Male Lead Falls in Love First",
      "votes": 1,
      "votes_plus": 1,
      "votes_minus": 0,
      "added_by": 32676484550
    },
    {
      "series_id": 45702686526,
      "category": "Male to Female",
      "votes": 2,
      "votes_plus": 2,
      "votes_minus": 0,
      "added_by": 10402087254
    },
    {
      "series_id": 45702686526,
      "category": "Monster/s",
      "votes": 9,
      "votes_plus": 9,
      "votes_minus": 0,
      "added_by": 70020080814
    },
    {
      "series_id": 45702686526,
      "category": "Multiple Persons In Same Body",
      "votes": 2,
      "votes_plus": 1,
      "votes_minus": 1,
      "added_by": 53789420519
    },
    {
      "series_id": 45702686526,
      "category": "Multiple Protagonists",
      "votes": 9,
      "votes_plus": 9,
      "votes_minus": 0,
      "added_by": 1966803365
    },
    {
      "series_id": 45702686526,
      "category": "Murder/s",
      "votes": 1,
      "votes_plus": 1,
      "votes_minus": 0,
      "added_by": 2095673730
    },
    {
      "series_id": 45702686526,
      "category": "Mysterious Location",
      "votes": 9,
      "votes_plus": 9,
      "votes_minus": 0,
      "added_by": 1966803365
    },
    {
      "series_id": 45702686526,
      "category": "Near-Death Experience",
      "votes": 4,
      "votes_plus": 3,
      "votes_minus": 1,
      "added_by": 54489895811
    },
    {
      "series_id": 45702686526,
      "category": "Netorare",
      "votes": 3,
      "votes_plus": 3,
      "votes_minus": 0,
      "added_by": 5601366475
    },
    {
      "series_id": 45702686526,
      "category": "nudity",
      "votes": 11,
      "votes_plus": 11,
      "votes_minus": 0,
      "added_by": 17832369998
    },
    {
      "series_id": 45702686526,
      "category": "Orphan/s",
      "votes": 9,
      "votes_plus": 9,
      "votes_minus": 0,
      "added_by": 70020080814
    },
    {
      "series_id": 45702686526,
      "category": "Post-Apocalyptic",
      "votes": 12,
      "votes_plus": 12,
      "votes_minus": 0,
      "added_by": 1966803365
    },
    {
      "series_id": 45702686526,
      "category": "Pregnancy",
      "votes": 3,
      "votes_plus": 3,
      "votes_minus": 0,
      "added_by": 2095673730
    },
    {
      "series_id": 45702686526,
      "category": "Prostitute/s",
      "votes": 2,
      "votes_plus": 1,
      "votes_minus": 1,
      "added_by": 2095673730
    },
    {
      "series_id": 45702686526,
      "category": "Psychopath/s",
      "votes": 1,
      "votes_plus": 1,
      "votes_minus": 0,
      "added_by": 2095673730
    },
    {
      "series_id": 45702686526,
      "category": "Quirky Character/s",
      "votes": 3,
      "votes_plus": 3,
      "votes_minus": 0,
      "added_by": 44221078641
    },
    {
      "series_id": 45702686526,
      "category": "Rape",
      "votes": 13,
      "votes_plus": 13,
      "votes_minus": 0,
      "added_by": 17832369998
    },
    {
      "series_id": 45702686526,
      "category": "Revenge",
      "votes": 0,
      "votes_plus": 0,
      "votes_minus": 0,
      "added_by": 2095673730
    },
    {
      "series_id": 45702686526,
      "category": "Romantic Subplot",
      "votes": 4,
      "votes_plus": 4,
      "votes_minus": 0,
      "added_by": 54489895811
    },
    {
      "series_id": 45702686526,
      "category": "Scar/s",
      "votes": 1,
      "votes_plus": 1,
      "votes_minus": 0,
      "added_by": 32676484550
    },
    {
      "series_id": 45702686526,
      "category": "Strong Male Lead",
      "votes": 4,
      "votes_plus": 3,
      "votes_minus": 1,
      "added_by": 54489895811
    },
    {
      "series_id": 45702686526,
      "category": "Subtle Romance",
      "votes": 2,
      "votes_plus": 2,
      "votes_minus": 0,
      "added_by": 54489895811
    },
    {
      "series_id": 45702686526,
      "category": "Suicide/s",
      "votes": 1,
      "votes_plus": 1,
      "votes_minus": 0,
      "added_by": 2095673730
    },
    {
      "series_id": 45702686526,
      "category": "Super Powers",
      "votes": 9,
      "votes_plus": 9,
      "votes_minus": 0,
      "added_by": 44221078641
    },
    {
      "series_id": 45702686526,
      "category": "Survival",
      "votes": 11,
      "votes_plus": 11,
      "votes_minus": 0,
      "added_by": 70020080814
    },
    {
      "series_id": 45702686526,
      "category": "Traumatic Past",
      "votes": 1,
      "votes_plus": 1,
      "votes_minus": 0,
      "added_by": 32676484550
    },
    {
      "series_id": 45702686526,
      "category": "Travel",
      "votes": 2,
      "votes_plus": 2,
      "votes_minus": 0,
      "added_by": 54489895811
    }
  ],
  "latest_chapter": 65,
  "forum_id": 12646704033,
  "status": "10 Volumes (Ongoing)",
  "licensed": true,
  "completed": false,
  "anime": {
    "start": "Vol 1, Chap 1",
    "end": "Vol 7, Chap 39"
  },
  "related_series": [],
  "authors": [
    {
      "name": "ISHIGURO Masakazu",
      "author_id": 2137592267,
      "type": "Author"
    },
    {
      "name": "ISHIGURO Masakazu",
      "author_id": 2137592267,
      "type": "Artist"
    }
  ],
  "publishers": [
    {
      "publisher_name": "Denpa",
      "publisher_id": 2681528803,
      "type": "English",
      "notes": "7 Volumes - Ongoing"
    },
    {
      "publisher_name": "Kodansha",
      "publisher_id": 52396899975,
      "type": "Original",
      "notes": ""
    }
  ],
  "publications": [
    {
      "publication_name": "Afternoon",
      "publisher_name": "Kodansha",
      "publisher_id": 52396899975
    }
  ],
  "recommendations": [],
  "category_recommendations": [
    {
      "series_name": "Usagi Yojimbo",
      "series_id": 66318217691,
      "weight": 22869
    },
    {
      "series_name": "Hidamari no Ki",
      "series_id": 55654128460,
      "weight": 21587
    },
    {
      "series_name": "Siren (Getty)",
      "series_id": 59233462697,
      "weight": 20640
    },
    {
      "series_name": "Kyoushirou 2030",
      "series_id": 77767402621,
      "weight": 20100
    },
    {
      "series_name": "Gajuu",
      "series_id": 11097959728,
      "weight": 20050
    }
  ],
  "rank": {
    "position": {
      "week": 531,
      "month": 899,
      "three_months": 1003,
      "six_months": 1035,
      "year": 750
    },
    "old_position": {
      "week": 595,
      "month": 899,
      "three_months": 676,
      "six_months": 791,
      "year": 437
    },
    "lists": {
      "reading": 876,
      "wish": 403,
      "complete": 15,
      "unfinished": 20,
      "custom": 196
    }
  },
  "last_updated": {
    "timestamp": 1724267067,
    "as_rfc3339": "2024-08-21T12:04:27-07:00",
    "as_string": "August 21st, 2024 12:04pm PDT"
  }
}
```

**Comment by diogovalentte** [comment]
Hi, I didn't understand exactly how to integrate it with Mantium. Can you explain more? A practical example would be great.

**Comment by diogovalentte** [comment]
Thank you very much for the images, they explain well how it would integrate into the interface. But how will Mantium track this manga?

After Mantium searches a list of mangas using the mangaupdates API and the user selects a manga within this list, what should Mantium do with this manga info from the mangaupdates API?

I see a "Get Chapters" button on the image you provided, would Mantium get the chapters list from the mangaupdates API? Does it make them available with links to the actual chapters, so that the user can click on the links in the Mantium dashboard and be redirected to the chapter to read it?

**Comment by diogovalentte** [comment]
Mantium's purpose is tracking manga from different sources, mangaupdates would just be another source, so I don't see a problem with adding it to Mantium.

The mangaupdates API has a route to get a feed with all chapters of the manga, so I can create the chapters list from this. The user will select the last read chapter from this list, and Mantium will be able to get the latest released chapter too. The chapters' links will redirect to the manga page on mangaupdates. It won't show the date of the last released chapter though, seems like it's not returned by any API route. I can show the release group instead.

The only problem is adding a manga to Mantium. The manga ID used by the API is not in the manga URL on the mangaupdates site, so I would need to do like in your images, getting the list of mangas and displaying it so that the user can select one of them.

Searching instead of providing the manga URL was my original thought for Mantium, but the library I used to create the dashboard (called streamlit) didn't have a good option to do it when I created Mantium, but now it has one.

There are some other things I plan to add to mantium, and I mostly work on the project on weekends, so I think I can add the mangaupdates source in one or two weeks from now.

**Comment by diogovalentte** [comment]
@ShrinkWrapper I released [Mantium v1.0.0](https://github.com/diogovalentte/mantium/releases/tag/v1.0.0) with the mangaupdates source and other changes.

There is a limitation with the mangaupdates source though. The chapters are listed by upload date in Mantium. This means that if a group uploads chapter 2 from a manga that another group already uploaded 50 chapters previously, chapter 2 will be considered the last chapter released. This is a limitation of MangaUpdates that can't properly sort the chapters by chapter number, so I had to use release date instead.
